scripts/add_electricity.py

```diff
@@ -368,54 +368,69 @@ def attach_wind_and_solar(
             else:
                 caps = pd.Series(index=ds.indexes["bus"]).fillna(0)
 
-            n.madd(
-                "Generator",
-                ds.indexes["bus"],
-                " " + tech,
-                bus=ds.indexes["bus"],
-                carrier=tech,
-                p_nom=caps,
-                p_nom_extendable=tech in extendable_carriers["Generator"],
-                p_nom_min=caps,
-                p_nom_max=ds["p_nom_max"].to_pandas(),
-                p_max_pu=ds["profile"].transpose("time", "bus").to_pandas(),
-                weight=ds["weight"].to_pandas(),
-                marginal_cost=costs.at[suptech, "marginal_cost"],
-                capital_cost=capital_cost,
-                efficiency=costs.at[suptech, "efficiency"],
-            )
+            if tech != "csp":
+                n.madd(
+                    "Generator",
+                    ds.indexes["bus"],
+                    " " + tech,
+                    bus=ds.indexes["bus"],
+                    carrier=tech,
+                    p_nom=caps,
+                    p_nom_extendable=tech in extendable_carriers["Generator"],
+                    p_nom_min=caps,
+                    p_nom_max=ds["p_nom_max"].to_pandas(),
+                    p_max_pu=ds["profile"].transpose("time", "bus").to_pandas(),
+                    weight=ds["weight"].to_pandas(),
+                    marginal_cost=costs.at[suptech, "marginal_cost"],
+                    capital_cost=capital_cost,
+                    efficiency=costs.at[suptech, "efficiency"],
+                )
 
-            if tech == "csp":  # add store and link for CSP
-                csp_buses_i = n.madd(
+            elif tech == "csp":  # add store and link for CSP
+                n.madd(
```

Comment:
Hi @davide-f and @katia,
Based on our last discussion to try to add CSP bus to network in `add_electricity`,  while doing this, I encountered an error with this section.

`ValueError: Length of values (42) does not match length of index (161)`
I investigated further. `ds["x"]` has 42 items, `ds["y"]` has 34 items, `ds["bus"]` has 161 items.

This is where I made mentioned that the cordinates were not tallying. I checked profiles for solar and onwind, it has similar values, it just that the workflow doesn't add them in `add_electricity`.

I think it is justifiable to go ahead with the second approach which is to add the generators first, then add the other components in `add_extra_components`.

What are you general thoughts on this?